### ISSUE TITLE
Rerunning pylint on the same code can produce inconsistent `import-self` warnings

### ISSUE THREAD

**Issue by sami-evangelista** [pylint-dev/pylint#10025]
### Bug description

I have this directory structure:
```
|-pack
| `-__init__.py
`-scripts
  `-pack
```

In `__init__.py`:
```python
"""A package to do stuffs."""

X = 1
```

In `pack`:
```python
"""A script to print X."""

from pack import X

print(X)
```

### Command used

```
pylint pack/ scripts/pack
```

### Pylint output

Output is inconsistent. It alternates between:
```
------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```
and:
```
************* Module pack
scripts/pack:3:0: W0406: Module import itself (import-self)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 6.67/10, +0.00)
```

### Expected behavior

`import-self` should not be raised.

### Pylint version

pylint 3.0.2
astroid 3.0.1
Python 3.11.2 (main, Sep 14 2024, 03:00:30) [GCC 12.2.0]

### OS / Environment

Debian 12
